packages/uploadthing/src/sdk/index.ts

```diff
@@ -256,7 +287,9 @@ export class UTApi {
       },
     );
 
-    const uploadFileResponse = Array.isArray(urls) ? uploads : uploads[0];
+    const uploadFileResponse = Array.isArray(urls)
+      ? [...errors, ...uploads]
+      : [...errors, ...uploads][0];
```

Comment:
Would be nice if order was preserved

Reply:
Would require infra changes, and I'm not sure how the UploadThingError being passed over the wire would go

Reply:
Would it? We'd just need to keep track of what indexes failed?